src/endpoints/transactions/transaction.service.ts

```diff
@@ -227,7 +312,7 @@ export class TransactionService {
     }
   }
 
-  async processTransactions(transactions: Transaction[], withScamInfo: boolean, assets?: Record<string, AccountAssets>): Promise<void> {
+  async processTransactions(transactions: Transaction[], withScamInfo: boolean, withUsername: boolean): Promise<void> {
```

Comment:
processTransactions can have as a parameter an `DefaultTransactionQueryOptions`, this class can be also.
This way it will be more clear that the first bool value is for ScamInfo and the 2nd one for Username. Also this way also the developer error will be removed, since he won`t be able anymore to mess up the order of the two adjacent parameters with the same type.

`TransactionQueryOptions` can also extend this class the default query options can be kept separate.

Reply:
created an options object also in this situation